src/main.ps1

```diff
@@ -0,0 +1,17 @@
+﻿$arch = ([string][System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture).ToLower()
+Write-Verbose "OS Architecture: $arch"
+$libSodiumVersion = '1.0.19.2'
+if ($IsWindows) {
+    $libPath = Join-Path $PSScriptRoot "\libs\libsodium\$libSodiumVersion\win-$arch\native\libsodium.dll"
+}
+if ($IsLinux) {
+    $libPath = Join-Path $PSScriptRoot "\libs\libsodium\$libSodiumVersion\linux-$arch\native\libsodium.so"
+}
+if ($IsMacOS) {
+    $libPath = Join-Path $PSScriptRoot "\libs\libsodium\$libSodiumVersion\osx-$arch\native\libsodium.dylib"
+}
+Write-Verbose "Library Path: $libPath"
+$null = [System.Runtime.InteropServices.NativeLibrary]::Load($libPath)
+
+$corePath = Join-Path $PSScriptRoot '\libs\Sodium.Core\1.3.5\Sodium.Core.dll'
+$null = [System.Reflection.Assembly]::LoadFile($corePath)
```

Comment:
Loading Sodium.Core in the default AssemblyLoadContext greatly simplifies the code, but causes conflicts if different modules attempt to load different versions of Sodium.Core.dll.  That is why I implemented an IsolatedAssemblyLoadContext in my PR.

Reply:
Loading the assembly into module context using NestedModules, but that is not covered here. Its covered in the other PR. Closing this for now.